### PR TITLE
Support for files

### PR DESCRIPTION
**Pull request by gstro**
Implements Read / Write for files using the sodium streaming interface.
- Computes MD5 during encryption so that it only requires one pass through the file
- Uses `FileHandler` interface for working with the file system
- Adds guards for `CommonCrypto` importing for reasons described below

The additions require Apple's `CommonCrypto` library as a module map following the instructions for framework projects [here](https://spin.atomicobject.com/2015/02/23/c-libraries-swift/). This will require users of this SDK to follow the same instructions for their own apps when integrating the SDK. This will be unnecessary when upgrading to XCode 10 / Swift 4.2, but currently [swift-sodium is not updated](https://github.com/jedisct1/swift-sodium/issues/155) for that yet. 

Additional tests and documentation to follow, just wanted to get a PR in for early review.

### CODE REVIEW

**Comment by gstro** [comment]
Ok, tests and doc comments added. Ready for review!

File: Example/Tests/CryptoTests.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            let input  = try FileHandle(forReadingFrom: dstUrl)
+            let buffer = input.readDataToEndOfFile()
+            input.closeFile()
+            XCTAssertEqual(buffer, data)
+        } catch {
+            return XCTFail(error.localizedDescription)
+        }
+    }
+
+    func testMD5() {
```
Comment (lkwdwrd) [inline Example/Tests/CryptoTests.swift]: Is this test looking at the right thing? I looks like this is the only call to Crypto.md5

File: Example/Tests/IntegrationTests.swift [hunk 2]
```diff
@@ -667,6 +667,266 @@ class IntegrationTests: XCTestCase, TestUtils {
...
+        let query = QueryParams(includeData: true, includeAllWriters: true)
+        asyncTest(#function + "read first") { (expect) in
+            shareClient.query(params: query) { (result) in
+                XCTAssertNotNil(result.value)
+                XCTAssert(result.value!.records.count == 0)
+                expect.fulfill()
+            }
+        }
+
+        // shared client should now see file contents
```
Comment (lkwdwrd) [inline Example/Tests/IntegrationTests.swift]: s/now/not/ ?

File: E3db/Classes/Crypto.swift [hunk 3]
```diff
@@ -3,16 +3,21 @@
 //  E3db
 //
 
+#if canImport(CommonCrypto)
```
[the remark this answers is not in the document]
Comment (gstro) [reply E3db/Classes/Crypto.swift]: Yeah, it's also related to the comment mentioned above -- it requires devs to incorporate the library in a particular way unless updated to XCode 10.

File: E3db/Classes/Crypto.swift [hunk 4]
```diff
@@ -164,3 +169,164 @@ extension Crypto {
...
+            throw E3dbError.cryptoError("Failed to initialize stream")
+        }
+        guard let dst = FileManager.tempBinFile() else {
+            throw E3dbError.cryptoError("Failed to open file")
+        }
+        let input  = try FileHandle(forReadingFrom: src)
+        let output = try FileHandle(forWritingTo: dst)
+
+        // manage resources
+        let context = UnsafeMutablePointer<CC_MD5_CTX>.allocate(capacity: 1)
```
[the remark this answers is not in the document]
Comment (gstro) [reply E3db/Classes/Crypto.swift]: Yeah, the function above isn't used during encryption, but it does help as a utility function, especially for testing. Could also be useful if needing to compute a checksum over a file for later files operations, potentially.

File: Example/Tests/CryptoTests.swift (hunk 1 above)
Comment (gstro) [reply Example/Tests/CryptoTests.swift]: Yep, just testing that the MD5 functionality is working as expected.

**Pull request merged**